doc/sphinx/proof-engine/ltac.rst

```diff
@@ -736,28 +745,28 @@ In some cases backtracking may be too expensive.
 
        assert_fails (first [ (idtac "1A" + idtac "1B" + fail) | idtac "2" ]; fail).
 
-   :tacn:`first` is an :token:`l1_tactic`.
-
    .. exn:: No applicable tactic.
       :undocumented:
 
-   .. todo the following is not accepted as a regular tactic but it does seem to do something
-      see https://github.com/coq/coq/pull/12103#discussion_r422249862.
-      Probably the same thing as for the :tacn:`solve` below.
-      The code is in Coretactics.initial_tacticals
+.. _taclist_in_first:
+
+   .. example:: Referring to a list of tactics in :cmd:`Tactic Notation`
 
-   .. tacn:: first [ {*| @ltac_expr } ]
+      This works similarly for the :tacn:`solve` tactic.
 
-      This is an |Ltac| alias that gives a primitive access to the first
-      tactical as an |Ltac| definition without going through a parsing rule. It
-      expects to be given a list of tactics through a :cmd:`Tactic Notation` command,
-      permitting notations with the following form to be written:
+    .. coqtop:: reset all
 
-      .. example::
+       Tactic Notation "myfirst" "[" tactic_list_sep(tacl,"|") "]" := first [ tacl ].
+       Goal True.
+       Fail myfirst [ auto | apply I ].  (* not working in 8.18 *)
 
-         .. coqtop:: in
+   .. note:: In :tacn:`first` and :tacn:`solve`, :n:`@ltac_expr`\s that don't
+      evaluate to tactic values are ignored.  So :tacn:`first` `[ () | 1 |` :tacn:`constructor` `]`
+      is equivalent to :tacn:`first` `[` :tacn:`constructor` `]`.
+      This may make it harder to debug scripts that inadvertently include non-tactic values.
```

Comment:
It's normal behavior in the sense that "Expression does not evaluate to a tactic" is caught by `first` as any other error from a tactic that fails to apply to the goal would be. Ltac makes no distinction between these different types of failures (I guess the only failure that would escape such a context would be an anomaly).

Similar example:

```coq
Goal True.
Fail (). (* Expression does not evaluate to a tactic *)
try 1. (* Success *)
```